### PR TITLE
Issue 21169 column type migration

### LINKED ISSUE #21169: Change datatype of date fields in MSSQL
On [this card](https://github.com/dotCMS/core/issues/20899) , the way dates were persisted was modified to consider time zones in Postgres. To finish this work, we need to do the same in MSSQL

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
The first take on this Upgrade task was incomplete it was ignoring a bunch of  object dependency errors thrown by the DB
I mistakenly thought that it was safe to ignore those errors but it turns out that they're actually needed. This new version adds all required scripts necessary for the migration to complete successfully

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task220402UpdateDateTimezonesTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    final String INSERT = "INSERT %s\n"
+            + "(id, language_code, country_code, [language], country, add_date)\n"
+            + "VALUES(%d, 'ES', 'CR', 'Spanish', 'Costa Rica', GetDate())";
+
+    final String DROP_TABLE = "drop table %s";
+
+    /**
+     * <b>Method to Test:</b> {@link Task220402UpdateDateTimezones#executeUpgrade()} <p>
+     * <b>Given Scenario:</b> When ms-sql is used, the upgrade task should be executed <p>
+     * <b>Expected Result:</b> When using postgres, dates should be declared as timestamps with
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task220402UpdateDateTimezonesTest.java:49]: it should be "when using mssql", right?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task220402UpdateDateTimezonesTest.java]: yeah

File: dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java [hunk 2]
```diff
@@ -546,19 +546,26 @@ private List<String> getColumnConstraintsMSSQL (final Connection connection, fin
     }
 
     public void dropIndex(final String tableName, final String indexName) throws SQLException {
+        try (Connection conn = DbConnectionFactory.getConnection()) {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java:549]: Is this try-with-resources needed ?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java]: yeah. After what I learned today I guess it is not.

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**